packages/rum-core/src/domain/rumEventsCollection/view/trackScrollMetrics.ts

```diff
@@ -0,0 +1,40 @@
+import type { Duration } from '@datadog/browser-core'
+import { throttle, addEventListener, DOM_EVENT } from '@datadog/browser-core'
+
+export interface ScrollMetrics {
+  maxScrollDepth?: number
+  maxscrollHeight?: number
+  maxScrollDepthTimestamp?: Duration
+}
+
+export function trackScrollMetrics() {
+  const scrollMetrics: ScrollMetrics = {}
+
+  const handleScrollEvent = throttle(
+    (event: Event) => {
+      const scrollTop =
+        window.pageYOffset || (document.documentElement || document.body.parentNode || document.body).scrollTop
+
+      const scrollHeight = Math.round(document.documentElement.scrollHeight)
+      const scrollDepth = Math.round(Math.min(document.documentElement.clientHeight + scrollTop, scrollHeight))
+
+      if (scrollDepth > (scrollMetrics?.maxScrollDepth ?? 0)) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏avoid using optional chaining operators when possible, because it is known to be verbose once transpiled. This condition is transpiled as:

```js
scrollDepth > ((_a = scrollMetrics === null || scrollMetrics === void 0 ? void 0 : scrollMetrics.maxScrollDepth) !== null && _a !== void 0 ? _a : 0)
```

Here,`scrollMetrics` is always defined, so you you can remove the optional chaining operator. Also, while we are at it, `??` has the same effect as `||` here, so you can rewrite it to:

```suggestion
      if (scrollDepth > (scrollMetrics.maxScrollDepth || 0)) {
```

which produces a much smaller transpiled output.